client/verta/verta/runtime.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+"""Classes and functions to enable context logging within predictions made against
+models on the Verta Platform.
+
+.. versionadded:: 0.22.0
+
+"""
+
+import json
+import threading
+from typing import Any, Dict, Optional
+
+
+_THREAD = threading.local()
+
+def _init_thread_logs() -> None:
+    """
+    Initialize our thread-local variable for storing logging context.
+    """
+    if not hasattr(_THREAD, 'logs'):
+        _THREAD.logs = dict()
+
+
+def _get_thread_logs() -> Dict[str, Any]:
+    """
+    Return the current thread-local logs or initialize a new dict.
+    """
+    _init_thread_logs()
+    return _THREAD.logs
+
+
+def _set_thread_logs(logs: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Sets the thread-local logs, overwriting any existing values.
+    """
+    _THREAD.logs = logs
+    return _THREAD.logs
+
+
+def _init_validate_flag() -> None:
+    """
+    Initialize our thread-local variable for the validation flag.
+    """
+    if not hasattr(_THREAD, 'validate'):
+        _THREAD.validate = False
+
+
+def _get_validate_flag() -> bool:
+    """
+    Return the current thread-local variable for validate or initialize a
+    new boolean.
+    """
+    _init_validate_flag()
+    return _THREAD.validate
+
+
+def _set_validate_flag(flag: bool) -> bool:
+    """
+    Sets the thread-local validate flag boolean value.
+    """
+    _THREAD.validate = flag
+    return _THREAD.validate
+
+
+def validate_json(value: Any) -> str:
+    """
+    Attempt to serialize the provided value to JSON.
+    Raise an error on failure.
+    """
+    try:
+        return json.dumps(value)
+    except json.JSONDecodeError as json_err:
+        raise Exception("Unable to convert logging data to JSON") from json_err
+    except TypeError as type_err:
+        raise Exception("Unable to serialize log value to JSON.") from type_err
+
+
+def log(key: str, value: Any) -> None:
+    """
+    Updates current logging context dict with provided key and value.
+    For use within a model's ``predict()`` method to collect logging
+    context, which is stored in thread-local variables.
+
+    Parameters
+    ----------
+    key : str
+        String value to use as data label (key) in logging context.
+    value : Any
+        Any JSON serializable value you wish to include in the context logs.
+
+    Returns
+    -------
+    None
+
+    Examples
+    --------
+    .. code-block:: python
+       :emphasize-lines: 10,13
+
+        from verta import runtime
+
+        # Sample model code:
+        class MyModel(VertaModelBase):
+            def __init__(self, artifacts):
+                "ok"
+
+            def predict(self, data):
```

Comment:
I know this is a bit irrelevant, but I'd like to help ingrain it as a habit for everyone 😬 

```suggestion
            @verify_io
            def predict(self, data):
```

Reply:
Good call.